agent/src/main/resources/build-scan-init.gradle

```diff
@@ -5,11 +5,43 @@ if (GradleVersion.current() < GradleVersion.version('4.1')) {
     return
 }
 
+initscript {
+    def geVersion = System.getProperty('teamCityBuildScanPlugin.gradle.enterprise.plugin.version')
+    def atLeastGradle5 = GradleVersion.current() >= GradleVersion.version('5.0')
+    def ccudVersion = System.getProperty('teamCityBuildScanPlugin.ccud.version')
+
+    if (geVersion && !gradle.getParent()) {
+        repositories {
+            maven { url 'https://plugins.gradle.org/m2' }
+        }
+        dependencies {
+            classpath atLeastGradle5 ?
+                    "com.gradle:gradle-enterprise-gradle-plugin:$geVersion" :
+                    "com.gradle:build-scan-plugin:$geVersion"
+
+            if (atLeastGradle5 && ccudVersion) {
```

Comment:
I would do this check not nested. This way, you could still apply the CCUD plugin even if the GE plugin is not defined as a config param (for scenarios where the GE plugin is already defined in the build).